server/athenian/api/align/goals/dbaccess.py

```diff
@@ -265,11 +265,22 @@ async def get_goal_template_from_db(template_id: int, sdb: DatabaseLike) -> Row:
     return template
 
 
-async def get_goal_templates_from_db(account: int, sdb: DatabaseLike) -> list[Row]:
+async def get_goal_templates_from_db(
+    account: int,
+    sdb: DatabaseLike,
+    exclude_with_metric_params: bool,
+) -> list[Row]:
     """Return all account GoalTemplate-s, ordered by id."""
-    stmt = (
-        sa.select(GoalTemplate).where(GoalTemplate.account_id == account).order_by(GoalTemplate.id)
-    )
+    where = GoalTemplate.account_id == account
+    from sqlalchemy import JSON
+
+    if exclude_with_metric_params:
+        # select both SQL NULL and JSON null
+        where = sa.and_(
+            where,
+            sa.or_(GoalTemplate.metric_params.is_(None), GoalTemplate.metric_params == JSON.NULL),
+        )
+    stmt = sa.select(GoalTemplate).where(where).order_by(GoalTemplate.id)
```

Comment:
Note: `ORDER BY` doesn't work with `UNION [ALL]` directly, this should save you some debugging time :slightly_smiling_face: